src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,161 @@
+import {
+    readableTimeCellRenderer,
+    sizeCellRenderer,
+    itemNameCellRenderer,
+} from 'box-ui-elements/es/features/virtualized-table-renderers';
+import VirtualizedTable from 'box-ui-elements/es/features/virtualized-table';
+import { Column } from 'react-virtualized/dist/es/Table/index';
+import { TABLE_COLUMNS } from './constants';
+
+const { KEY_NAME, KEY_MODIFIED_AT } = TABLE_COLUMNS;
+
+class ArchiveExplorer extends React.Component {
+    static propTypes = {
+        itemCollection: PropTypes.arrayOf(
+            PropTypes.shape({
+                type: PropTypes.string.isRequired,
+                absolute_path: PropTypes.string.isRequired,
+                name: PropTypes.string.isRequired,
+                modified_at: PropTypes.string.isRequired,
+                size: PropTypes.number.isRequired,
+                path_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+                parent: PropTypes.string,
+                item_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string.isRequired,
+                            name: PropTypes.string,
+                        }),
+                    ).isRequired,
+                }).isRequired,
+            }),
+        ).isRequired,
+    };
+
+    buildCache = memoize(itemList => {
+        const folders = [];
+        const temp = {};
+        const cache = {};
+        let root;
+
+        itemList.forEach(info => {
+            if (!info.parent) {
+                root = info.absolute_path;
+            }
+            if (info.type === 'folder') {
+                folders.push(info);
+            }
+            temp[info.absolute_path] = info;
+        });
+
+        folders.forEach(folderInfo => {
+            cache[folderInfo.absolute_path] = folderInfo.item_collection.entries.map(item => temp[item.absolute_path]);
+        });
+
+        return { cache, root };
+    });
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            fullPath: undefined,
+        };
+    }
+
+    getRowData = itemList => ({ index }) => {
+        const { name, type, modified_at: modifiedAt, ...rest } = itemList[index];
+
+        const rowData = {
+            [KEY_NAME]: {
+                isExternal: false,
+                name,
+                type,
+            },
+            [KEY_MODIFIED_AT]: `20${modifiedAt}`,
```

Comment:
What does the 20 signify in this statement?

Reply:
It's just a temporary way to fix JS date parsing confusion (make '19' to '2019'). Conversion will fix it later.